bing_images_api.py

```diff
@@ -17,61 +17,78 @@ def download(url, file_name):
-        "license": "public",
-        "imageType": "photo",
-        'minHeight': 100,
-        'minWidth': 100,
-        # +using count_step to determine # of URLs instead of range
-        'count': image_count,
-        'market': 'en-us'
-    }
-    #Variables and calculation for offsets (150 max images, 35 per page)
-    split=image_count/150
-    base=1
-    z=0
-    listurl = []
-    #Requests for ContentURL, loop to capture each URL
-    #res=response, search_results=list/dictionary of info, url=one captured URL, listurl=stored URLS from the search_term
-    while z<split:
-        params.update({'offset': z*150})
-        res = requests.get("https://api.cognitive.microsoft.com/bing/v7.0/images/search?q=", headers=headers, params=params)
-        search_results = res.json()
-        i=0
-        while i<len(search_results['value']):
-            url = search_results['value'][i]['contentUrl']
-            listurl.append(url)
-            print("adding url:"+ url)
-            i=i+1
-        z=z+1
-        # looks for outfile, if none exists one is created
-        #the list of urls is then appended to the old list of urls (if one exists)
-        #prevlisturl=urls stored in the outfile, listurl=current url list
-        try:
-            with open('outfile{}'.format(term), 'rb') as fp:
-                prevlisturl = pickle.load(fp)
-                prevlisturl.append(listurl)
-                set(prevlisturl)
-                pickle.dump(prevlisturl, fp)
-        except:
-            with open('outfile{}'.format(term), 'wb') as fp:
-                pickle.dump(listurl, fp)
-#downloads the list stored in the outfile, removes duplicate urls as well
-#file_name=name to call images, search_location= save location path
+    for val in search_terms:
+        subscription_key = "89606185ba57483ab3952f85df1b39bb"
+        headers = {"Ocp-Apim-Subscription-Key": subscription_key}
+        params = {
+            "q": val,
+            # +define the term for offset
+            "offset": 0,
+            "imageType": "photo",
+            'minHeight': 100,
+            'minWidth': 100,
+            # +using count_step to determine # of URLs instead of range
+            'count': image_count,
+            'market': 'en-us'
+        }
+        #Variables and calculation for offsets (150 max images, 35 per page)
+        base = int(image_count/150)
+        z = 0
+        a = 0
+        listurl = []
+        #Requests for ContentURL, loop to capture each URL
+        #res=response, search_results=list/dictionary of info, url=one captured URL, listurl=stored URLS from the search_terms
+        while z < base:
+            params.update({'offset': a})
+            res = requests.get("https://api.cognitive.microsoft.com/bing/v7.0/images/search?q=", headers=headers, params=params)
+            search_results = res.json()
+            i=0
+            if search_results.get('value'):
+                while i < len(search_results['value']):

```

Comment:
use a for loop. I will show you.

Reply:
Traceback (most recent call last):
  File "B:/Libraries/Google Drive/Pycharm/Github/aicamp/bing_images_api.py", line 130, in <module>
    save_url(search_terms, image_count, term)
  File "B:/Libraries/Google Drive/Pycharm/Github/aicamp/bing_images_api.py", line 73, in save_url
    prev_list_url = set(prev_list_url)
TypeError: unhashable type: 'list'


error needs to  be fixed for the with open statement